TypeCobol/Compiler/CodeElements/CodeElement.cs

```diff
@@ -157,16 +159,44 @@ public override string ToString()
 
         public override bool Equals(object obj)
         {
-            var codeElement = obj as CodeElement;
-            if (codeElement == null)
-                return false;
-
-            return this.Line == codeElement.Line &&
-                   this.Type == codeElement.Type &&
-                   this.Column == codeElement.Column &&
-                   this.StartIndex == codeElement.StartIndex &&
-                   this.StopIndex == codeElement.StopIndex &&
-                   this.Text == codeElement.Text;
+            return Equals(obj as CodeElement);
+        }
+
+        public bool Equals(CodeElement codeElement)
+        {
+            if (Object.ReferenceEquals(this, codeElement)) return true;
+            if (Object.ReferenceEquals(null, codeElement)) return false;
+
+            if (ConsumedTokens != null && codeElement.ConsumedTokens != null)
+            {
+                if (ConsumedTokens.Count > 0 && codeElement.ConsumedTokens.Count > 0)
+                {
+                    return Type == codeElement.Type && ConsumedTokens[0].Equals(codeElement.ConsumedTokens[0]);
+                }
+                else
+                {
+                    return Type == codeElement.Type && Object.ReferenceEquals(ConsumedTokens, codeElement.ConsumedTokens);
```

Comment:
If we reach this line, it means `this` and codeElement are 2 differents instances. 
Their ConsumedTokens contains no elements.
And their ConsumedTokens properties point to the same list. 

I wonder if it can happens.

I've tested to throw an exception and our tests run fine.

Reply:
About the comparison of CodeElements with no ConsumedToken, see issue #204. Properties based on ConsumedTokens collections have been modified to allow for null value of the collection.

Reply:
Equals, if ConsumedTokens are not both not_null, returns false;
Equals, if counts of ConsumedTokens are different, returns false.
Equals, if counts of ConsumedTokens are the same and superior to 0, returns equality of Type and ConsumedTokens[0].
Equals, if counts of ConsumedTokens are both equal to 0, throws an exception.
https://github.com/TypeCobolTeam/TypeCobol/pull/1587/commits/1be2e096b4f27bb53cd71184aacdcd855f430e29